cmd/govim/config/internal/applygen/main.go

```diff
@@ -0,0 +1,105 @@
+// applygen is a command that automates the generation of an Apply method on
+// the pointer receiver of a struct type which has exported pointer-type fields
+// to apply overrides from the argument onto the receiver
+package main
+
+import (
+	"bytes"
+	"flag"
+	"fmt"
+	"go/format"
+	"go/types"
+	"os"
+	"os/exec"
+	"sort"
+
+	"golang.org/x/tools/go/packages"
+)
+
+func main() {
+	os.Exit(main1())
+}
+
+func main1() int {
+	if err := mainerr(); err != nil {
+		if ee, ok := err.(*exec.ExitError); ok {
+			return ExitCode(ee.ProcessState)
+		}
+		fmt.Fprintln(os.Stderr, err)
+		return 1
+	}
+	return 0
+}
+
+func mainerr() error {
+	flag.Parse()
+
+	if len(flag.Args()) == 0 {
+		return fmt.Errorf("expected at least one argument")
+	}
+
+	cfg := &packages.Config{
+		Mode: packages.NeedFiles | packages.NeedSyntax | packages.NeedTypes | packages.NeedTypesInfo | packages.NeedName,
+	}
+	pkgs, err := packages.Load(cfg)
+	if err != nil {
+		return fmt.Errorf("failed to load types: %v", err)
+	}
+	if l := len(pkgs); l != 1 {
+		return fmt.Errorf("expected a single package; got %v", l)
+	}
+	toGen := make(map[string][]string)
+	var names []string
+	pkg := pkgs[0]
+	for _, name := range flag.Args() {
+		names = append(names, name)
+		tn := pkg.Types.Scope().Lookup(name)
+		if tn == nil {
+			return fmt.Errorf("failed to find type %v", name)
+		}
+		t := tn.(*types.TypeName).Type().Underlying()
+		st := t.(*types.Struct)
+		fields := make([]string, st.NumFields())
+		for i := 0; i < st.NumFields(); i++ {
+			fields[i] = st.Field(i).Name()
+		}
+		toGen[name] = fields
+	}
+	sort.Strings(names)
+
+	var buf bytes.Buffer
+
+	pf := func(format string, args ...interface{}) {
+		if format[len(format)-1] != '\n' {
```

Comment:
Any reason you aren't doing `format = strings.TrimSuffix(format, "\n") + "\n"` instead? (to avoid out of index panics, not that there is a risk for that here).

Reply:
This is just a simply crutch function here that avoids a bit of typing, not looking to be super advanced/useful. Keeps things a bit prettier later on. I could equally have done what you propose :)